### ISSUE TITLE
`cdktf get` Fails for AWS Provider version 4.67.0

### ISSUE THREAD

**Issue by stewartmoreland** [hashicorp/terraform-cdk#2879]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

version: cdktf v0.16.1
language: Python 3.10
providers:
  - aws = 4.67.0

### Affected Resource(s)

All provider jsii compliations using aws provider version 4.67.0.

### Debug Output

https://gist.github.com/stewartmoreland/b2352edddd6d7a2b4787b473cb199d0b

### Expected Behavior

`cdktf get` command should render provider input modules once transpiled from HCL to Python.

### Actual Behavior

Provider imports fail with `[ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created` occurring on various QuickSight resources.

```
providers/aws/appsync-domain-name/index.ts:79:14 - warning JSII5019: The property name "appsyncDomainName" conflicts with the declaring class "AppsyncDomainName". This will result in renaming the class to "_AppsyncDomainName" in C#. Consider renaming "appsyncDomainName".
  | 79   public get appsyncDomainName() {
  |                 ~~~~~~~~~~~~~~~~~
  |   providers/aws/appsync-domain-name/index.ts:34:14
  |     34 export class AppsyncDomainName extends cdktf.TerraformResource {
  |                     ~~~~~~~~~~~~~~~~~
  |     The declaring class is introduced here
  | providers/aws/quicksight-template/index-structs/structs400.ts:9180:14 - error JSII8007: Interface contains behavior. Rename "QuicksightTemplateDefinitionSheetsFilterControlsList" to "IQuicksightTemplateDefinitionSheetsFilterControlsList"
  | 9180 export class QuicksightTemplateDefinitionSheetsFilterControlsList extends cdktf.ComplexList {
```

### Steps to Reproduce

Set the `hashicorp/aws` provider version to `4.67.0` in your `cdktf.json` file. Remove existing imports and run `cdktf get`.

### Important Factoids

AWS provider version 4.66.1 still functions as expected and will be used until QuickSight resources can be transpiled successfully.

**Comment by yuriy-yarosh** [comment]
@stewartmoreland  Last Confirmed working AWS provider version is **4.66.1**.

**Comment by yuriy-yarosh** [comment]
Duplicated this issue for the [AWS Provider](https://github.com/hashicorp/terraform-provider-aws/issues/31615).
Newer AWS provider version is quite bloated compared to the previous releases.
We'll probably need some form of a workaround to simplify or trim imports - made a feature request #2892 , upvotes appreciated.

**Comment by mutahhir** [comment]
Thanks for creating the issue! Taking a look now.

**Comment by mutahhir** [comment]
Thanks for creating the issue! Taking a look now.

**Comment by xiehan** [comment]
This is resolved by upgrading to cdktf v0.17 and v15 of the AWS provider.

The prebuilt provider is also finally updated: cdktf/cdktf-provider-aws#1076 - note that some package managers may need an hour or two to pick up the new version.
